src/interfaces/IBLSApkRegistry.sol

```diff
@@ -33,6 +33,18 @@ interface IBLSApkRegistry is IRegistry {
         BN254.G2Point pubkeyG2;
     }
 
+    /**
+     * @notice Struct used to checkpoint the previous pubkeys and their hashes for a given operator
+     * @param previousPubkeyG1 is the G1 public key of the operator
+     * @param previoudPubkeyHash is the hash of the public key of the operator
+     * @param blockNumber is the block number at which the pubkey was updated
+     */
+    struct PubkeyCheckpoint {
+        BN254.G1Point previousPubkeyG1;
+        bytes32 previousPubkeyHash;
+        uint32 blockNumber;
```

Comment:
one edge case to consider is signatures on the boundary
e.g. if the operator signs a message with one key, changes keys, and signs a message with the new key, all within a single block.
the description you have looks technically accurate, and I feel like we probably just can't perfectly solve this problem? so if you care about historical keys (e.g. for slashing) you should just treat both keys as owned by the operator during this block, right?

Reply:
talked in-person, concluded this isn't possible because the existing check prevents deregistering and re-registering in the same block (which is the only way to change your key). specific code:
`lastDeregistrationTimestamp[operator] + deregistrationCooldown < block.timestamp`
note in particular the strict inequality